app/assets/javascripts/modules/highlight-active-section-heading.js

```diff
@@ -0,0 +1,136 @@
+;(function (Modules, root) {
+  'use strict'
+
+  var $ = root.$
+  var $window = $(root)
+
+  Modules.HighlightActiveSectionHeading = function () {
+    var self = this
+    var _hasResized = true
+    var _hasScrolled = true
+    var _interval = 50
+    var anchorIDs = []
+
+    self.getWindowDimensions = function () {
+      return {
+        height: $window.height(),
+        width: $window.width()
+      }
+    }
+
+    self.getWindowPositions = function () {
+      return {
+        scrollTop: $window.scrollTop()
+      }
+    }
+
+    self.getElementOffset = function ($el) {
+      return $el.offset()
+    }
+
+    self.start = function ($el) {
+      $window.resize(self.hasResized)
+      $window.scroll(self.hasScrolled)
```

Comment:
This is totally personal preference, but I've moved to using e.g. `.on('scroll', …)` as I think it makes it a million times clearer that you're setting up a new binding, vs telling the window to resize or scroll itself.